config/initializers/backend.rb

```diff
@@ -0,0 +1,5 @@
+require 'client_builder'
+
+Rails.application.config.http_client_uk = ClientBuilder.new('uk').call if TradeTariffFrontend::ServiceChooser.service_choices.present?
```

Comment:
Rather than putting the logic for whether to build a client into the initialiser, how about making `.call` check the service chooser `.service_choices` (it already is for hostname) - and return nil if there aren't any choices?